### PR TITLE
8328572: JFR: Use Class.forPrimitiveName(String)

### PR DESCRIPTION
**Pull request by egahlin**
Could I have a review of change that use Class.forPrimitiveName(String) instead checking against every primitive type.

Testing: jdk/jdk/jfr

Thanks
Erik

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328572](https://bugs.openjdk.org/browse/JDK-8328572): JFR: Use Class.forPrimitiveName(String) (**Enhancement** - P4)

### Reviewers
 * [Markus Grönlund](https://openjdk.org/census#mgronlun) (@mgronlun - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19235/head:pull/19235` \
`$ git checkout pull/19235`

Update a local copy of the PR: \
`$ git checkout pull/19235` \
`$ git pull https://git.openjdk.org/jdk.git pull/19235/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19235`

View PR using the GUI difftool: \
`$ git pr show -t 19235`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19235.diff">https://git.openjdk.org/jdk/pull/19235.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19235#issuecomment-2110839102)

### CODE REVIEW

File: src/jdk.jfr/share/classes/jdk/jfr/internal/util/Utils.java [hunk 1]
```diff
@@ -243,67 +243,24 @@ public static void ensureInitialized(Class<? extends jdk.internal.event.Event> e
...
-            return booleans;
-        case "java.lang.String":
-            String[] strings = new String[length];
-            for (int i = 0; i < length; i++) {
-                strings[i] = (String) values.get(i);
-            }
-            return strings;
+        Object array =  Array.newInstance(componentType, length);
+        for (int index = 0; index < length; index++) {
+            Array.set(array, index, values.get(index));
```
Comment (liach) [inline src/jdk.jfr/share/classes/jdk/jfr/internal/util/Utils.java:253]: Note that `Array.set` is slow (see [JDK-8051447](https://bugs.openjdk.org/browse/JDK-8051447)); yet this woudld be fine if this is not on hot code paths.

Comment (egahlin) [reply src/jdk.jfr/share/classes/jdk/jfr/internal/util/Utils.java:253]: It happens when a recording file is parsed, typically the Category annotation, which means about 150-200 invocation per file. Don't think it matters.

Comment (liach) [reply src/jdk.jfr/share/classes/jdk/jfr/internal/util/Utils.java:253]: Indeed, that sounds all good. If this becomes a performance issue, it should be fixed from java.lang.reflect instead.

**Review by mgronlun — APPROVED** [review]

**Pull request closed without merging**